server/src/main/java/org/apache/druid/server/QueryResultPusher.java

```diff
@@ -138,11 +144,17 @@ public Response push()
       // Response until it has consumed the underlying Sequence.
       asyncContext = request.startAsync();
       response = (HttpServletResponse) asyncContext.getResponse();
+      response.setHeader(HttpHeader.TRAILER.toString(), QueryResource.ERROR_MESSAGE_TRAILER_HEADER);
```

Comment:
You are setting error on the start?

Reply:
As per standard, we need to specify the header names which will be added as `Trailer`. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Trailer